src/services/publisher_service/helpers.rs

```diff
@@ -276,26 +279,10 @@ pub async fn dead_letters_check(
 }
 
 /// Checks for message is created more than threshold
-#[instrument(skip(postgres, metrics))]
-pub async fn dead_letter_give_up_check(
-    notification: Uuid,
+#[instrument]
+pub fn dead_letter_give_up_check(
+    notification_created_at: DateTime<Utc>,
     threshold: Duration,
-    postgres: &PgPool,
-    metrics: Option<&Metrics>,
-) -> std::result::Result<bool, sqlx::error::Error> {
-    let query_to_check = "
-        SELECT now() - created_at > interval '$1 seconds'
-        FROM subscriber_notification
-        WHERE id = $2
-    ";
-    let start = Instant::now();
-    let row: (bool,) = sqlx::query_as(query_to_check)
-        .bind(threshold.as_secs() as i64)
-        .bind(notification)
-        .fetch_one(postgres)
-        .await?;
-    if let Some(metrics) = metrics {
-        metrics.postgres_query("dead_letter_give_up_check", start);
-    }
-    Ok(row.0)
+) -> bool {
+    notification_created_at + threshold < Utc::now()
```

Comment:
This ordering is easier to understand to me than `Utc::now() - notification_created_at > threshold`

Order the variables in a timeline, oldest to newest.